packages/@aws-cdk/aws-elasticsearch/lib/domain.ts

```diff
@@ -1594,6 +1598,9 @@ export class Domain extends DomainBase implements IDomain {
     this.domain = new CfnDomain(this, 'Resource', {
       domainName: this.physicalName,
       elasticsearchVersion,
+      accessPolicies: new iam.PolicyDocument({
```

Comment:
I remember we intentionally didn't use this because of some limitation it presented. @stephanh Do you remember what it was?

Reply:
@iliapolo It would be great to understand the meaning (@stephanh). Anyway I tested it with `CfnDomain` and it was deployed and is working well.
However I'd like to use `aws-elasticsearch` lib and I hope we can improve and remove redundant code (piece of CF stack).

One more question - do you have any ETA on marking as stable (right now it has experimental status)?

Reply:
I think this is the issue we had around this https://github.com/aws/aws-cdk/pull/8369#discussion_r491905941.

Reply:
@dsharkou So the reason we did this is because we wanted to include the domain ARN in the resource policy. 

Something doesn't add up here, since the policy still contains `this.domainArn`, I would expect the [integration test](https://github.com/aws/aws-cdk/blob/master/packages/@aws-cdk/aws-elasticsearch/test/integ.elasticsearch.unsignedbasicauth.ts) to fail. Are you sure to ran it with the up to date code? 

In any case if we can just use `*` there instead, i'm also good with that. Once you get the build passing, i'm happy to merge this.  

Regarding the stability of the module, no ETA just yet, but we are starting to consider it. We will update in the [tracking issue](https://github.com/aws/aws-cdk/issues/6910).